### PR TITLE
SUBGRAPH: prove implications/anti-implications between 23 and other equations

### LINKED ISSUE #75: SUBGRAPH: Add the implications/anti-implications between 23 and the other equations as conjecture statements.
Note that 23 is dual to 8, so one should be able to adapt the implications involving 8.

### PR DESCRIPTION
**Pull request by ChienYungChi**
I skipped the step of putting conjectures and directly proved them.

Closes #75

### CODE REVIEW

**Comment by ChienYungChi** [comment]
await-review

**Comment by ChienYungChi** [comment]
awaiting-review

File: equational_theories/Subgraph.lean [hunk 1]
```diff
@@ -244,6 +256,25 @@ theorem Equation8_not_implies_Equation3 : ∃ (G : Type) (_ : Magma G), Equation
   · use 1
     simp only [Fin.isValue, one_ne_zero, not_false_eq_true]
 
+theorem Equation23_not_implies_Equation3 : ∃ (G : Type) (_ : Magma G), Equation23 G ∧ ¬ Equation3 G := by
+  simp only [not_forall]
+  use (Fin 2)
+  use ⟨(. + .)⟩
```
Comment (Command-Master) [inline equational_theories/Subgraph.lean:262]:
```suggestion
  use ⟨(· + ·)⟩
```

**Review by Command-Master — APPROVED** [review]

**Comment by pitmonticone** [comment]
Please merge from `main` and the CI checks will pass.

**Comment by pitmonticone** [comment]
LGTM. Thanks!

**Pull request merged**